modules/ROOT/pages/clauses/load-csv.adoc

```diff
@@ -397,4 +570,367 @@ If `LOAD CSV` is invoked with a `file:///` URL that points to your disk `file()`
+
+
+=== Database heap and memory
+
+To help handle larger volumes of transactions, there are a few memory configuration settings you can tweak:
+
+* link:{neo4j-docs-base-uri}/operations-manual/{page-version}/configuration/configuration-settings/#config_server.memory.heap.initial_size[`server.memory.heap.initial_size`] and link:{neo4j-docs-base-uri}/operations-manual/{page-version}/configuration/configuration-settings/#config_server.memory.heap.max_size[`server.memory.heap.max_size`]: set to at least 4G.
+* link:{neo4j-docs-base-uri}/operations-manual/{page-version}/configuration/configuration-settings/#config_server.memory.pagecache.size[`server.memory.pagecache.size`]: ideally, a value large enough to keep the whole database in memory.
+
+As a rule of thumb, you can create or update one million records in a single transaction per 2 GB of heap.
+
+
+== A full example
+
+.A full example
+====
+
+.Erase current database and import the full movie dataset
+[source, cypher]
+----
+// Clear data
+MATCH (n) DETACH DELETE n;
+
+// Create constraints
+CREATE CONSTRAINT Person_tmdbId IF NOT EXISTS
+FOR (p:Person)
+REQUIRE p.tmdbId IS UNIQUE;
+
+CREATE CONSTRAINT Movie_movieId IF NOT EXISTS
+FOR (m:Movie)
+REQUIRE m.movieId IS UNIQUE;
+
+// Create person nodes
+LOAD CSV FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS row
+MERGE (p:Person)
+SET
+p.tmdbId = toInteger(row.person_tmdbId),
+p.name = row.name,
+p.born = date(row.born);
+
+// Create movie nodes
+LOAD CSV FROM 'https://data.neo4j.com/importing-cypher/movies.csv' AS row
+MERGE (m:Movie)
+SET
+m.movieId = toInteger(row.movieId),
+m.title = row.title,
+m.imdbId = toInteger(row.movie_imdbId),
+m.languages = split(row.languages, '|'),
+m.genres = split(row.genres, '|');
+
+// Create relationships
+LOAD CSV FROM 'https://data.neo4j.com/importing-cypher/acted_in.csv' AS row
+MATCH (p:Person {tmdbId: toInteger(row.person_tmdbId)})
+MATCH (m:Movie {movieId: toInteger(row.movieId)})
+MERGE (p)-[r:ACTED_IN]->(m)
+SET r.role = row.role;
+
+// Set additional node label
+MATCH (p:Person)-[:ACTED_IN]->()
+WITH DISTINCT p SET p:Actor;
+----
+
+.Result
+[source, role="queryresult"]
+----
++-------------------+
+| No data returned. |
++-------------------+
+Added 2 constraints.
+Nodes created: 961
+Relationships created: 372
+Properties set: 3441
+Labels added: 1065
+----
+====
+
+== Other ways of importing data
+
+There are a few other tools to get CSV data into Neo4j.
 
+1. *The link:{neo4j-docs-base-uri}/operations-manual/{page-version}/tools/neo4j-admin/neo4j-admin-import/[`neo4j-admin database import`] command* is the most efficient way of importing large CSV files.
+2. Use a link:https://neo4j.com/docs/create-applications/[*language library*] to parse CSV data and run creation Cypher queries against a Neo4j database.
+3. link:https://neo4j.com/labs/apoc/[APOC^]: Awesome Procedures on Cypher.
```

Comment:
Please do not link APOC Extended in our official docs for Cypher 😭 Only Core, and even then, I would err on the side of "is it really really needed?"